src/app/home/home.component.ts

```diff
@@ -125,6 +127,24 @@ export class HomeComponent implements OnInit {
         this.currentTime = simulatedDate;
       }
     }
+
+    // start the redirect 12 hours in advance of CC open:
+    const startRedirectingToCCAt = new Date('2023-10-28T00:00:00+00:00');
+
+    // end the redirect exactly at the time CC closes.
+    if (
+      !queryParams.hasOwnProperty('noredirect') &&
+      this.currentTime >= startRedirectingToCCAt &&
+      this.currentTime < CCCloseDate &&
+      isPlatformBrowser(this.platformId)) {
+      this.router.navigate(
```

Comment:
we planned to do a 302 rather than a 301 redirect, but as this is a purely in-browser redirect the distinction isn't relevant. I'm fairly sure that browsers will treat this redirect as temporary.

Reply:
Right – it's not an HTTP redirect at all so browsers will add it to location history and would never repeat it without running the same JS again.